backend-tests/backend-tests.hs

```diff
@@ -460,27 +491,34 @@ tests :: FilePath -> TestTree
 tests wasm_file = testGroup "Tests" $ upgradeGroups $
   [ withoutUpgrade $ iiTest "installs" $ \ _cid ->
     return ()
+
   , withoutUpgrade $ iiTest "installs and upgrade" $ \ cid ->
     doUpgrade cid
+
   , withoutUpgrade $ iiTest "register with wrong user fails" $ \cid -> do
-    callIIReject cid dummyUserId #register (device1, invalidPOW)
+    callIIRejectExpectingMessage cid dummyUserId #register (device1, powAt cid 1) "[a-z0-9-]+ could not be authenticated against"
+
   , withoutUpgrade $ iiTest "register with bad pow fails" $ \cid -> do
-    callIIReject cid dummyUserId #register (device1, invalidPOW)
+    callIIRejectExpectingMessage cid webauthID #register (device1, invalidPOW) "proof of work hash check failed"
+
   , withoutUpgrade $ iiTest "register with future pow fails" $ \cid -> do
-    callIIReject cid dummyUserId #register (device1, powAt cid (20*60*1000_000_000))
+    callIIRejectExpectingMessage cid webauthID #register (device1, powAt cid (20*60*1000_000_000)) "proof of work timestamp 1200000000000 is too far in future, current time: 8"
```

Comment:
I think the current time of "8" is a bit too precise, since it can change very easily. Wdyt?
```suggestion
    callIIRejectExpectingMessage cid webauthID #register (device1, powAt cid (20*60*1000_000_000)) "proof of work timestamp 1200000000000 is too far in future, current time:"
```

Similar argument could be made for 1200... as well actually, but your call! (same goes for other test cases for timestamps, nonces)

Reply:
Done